### PR TITLE
Ask and clone a project

### PR DESCRIPTION
**Pull request by FintanH**
Adding a test to confirm the expected steps of asking for a RadUrn
and having to clone from the PeerId that responded.

### CODE REVIEW

**Review by kim — COMMENTED** [review]
This is basically the same as `providers_worksed`, isn't it?

Perhaps we should start gardening those integration tests a bit -- they're all very IO-bound and prone to become a steaming bowl of copy-pasta carbs.

**Comment by xla** [comment]
> Perhaps we should start gardening those integration tests a bit -- they're all very IO-bound and prone to become a steaming bowl of copy-pasta carbs.

Plus they are time sensitive. This PR was sparked because I had a hole in my understanding of when link automatically fetches, which I know understand is only when we track the `(project_urn, peer_id)` tuple. If we can collapse into `providers_works` - that be ideal.

**Comment by FintanH** [comment]
Yup! Happy to do that.

Another question I had in my head was: would this combined functionality be useful in the API?

**Comment by kim** [comment]
> Another question I had in my head was: would this combined functionality
> be useful in the API?

Ya I think this was once considered -- there is a whole bunch of timeouts and
errors to handle, though, so that would indeed be rather high-level. Perhaps
explore all of those, and how to surface them, first?

**Comment by xla** [comment]
> Another question I had in my head was: would this combined functionality be useful in the API?

As it is based on `providers` it mimics a request/response pattern, which in many scenarios is not valuable. For example the entire reason for radicle-dev/radicle-upstream#903 to be this ellaborate is to account for the fact that we get to know about an interesting `Have` well outside of the timeout of `providers`, maybe after restarts and so on. IMHO we should consider building out support for these scenarios in the protocol rather than give false hope of instant availability.

**Review by xla — APPROVED** [review]
🕍 🖱 🏫 😺

File: librad/tests/propagation_basic.rs [hunk 1]
```diff
@@ -384,3 +384,62 @@ async fn providers_works() {
...
+        let peer_id = match res {
+            Some(peer_info) => peer_info.peer_id,
+            None => panic!("Expected to have obtained peer1 but got None instead"),
+        };
+
+        assert!(!peer2.storage().has_urn(&repo_urn).unwrap());
+
+        {
+            let url = repo_urn.clone().into_rad_url(peer_id);
+            let git = peer2.storage().reopen().unwrap();
```
Comment (kim) [inline librad/tests/propagation_basic.rs:387]: Curious: what makes you think it should be reopened here?

Comment (FintanH) [reply librad/tests/propagation_basic.rs:387]: I need an owned `Storage` for the `move` in `spawn_blocking`. I actually looked at other tests to gauge what was best practice and found this https://github.com/radicle-dev/radicle-link/blob/master/librad/tests/propagation_basic.rs#L167-L179.

Do you see another way of getting around this?

**Review by NunoAlexandre — APPROVED** [review]
:small_airplane:

File: librad/tests/propagation_basic.rs (hunk 1 above)
Comment (kim) [reply librad/tests/propagation_basic.rs:387]: I see, yeah. Will be obsolete anyways with #327

**Pull request merged**